### ISSUE TITLE
How to change the amount of anchors in different detection layers of YOLOV3 ? 

### ISSUE THREAD

**Issue by JackSnowWolf** [pjreddie/darknet#709]
Since I want to deal with some other question by using YOLOV3, I need to change the amount of anchors in different detection layers. I have already change the filter number and num to corresponding value. However, when I change the amount of anchor to four or more than four, it shows me the error

Learning Rate: 0.001, Momentum: 0.9, Decay: 0.0001
Loaded: 0.495730 seconds
CUDA Error: invalid argument
CUDA Error: invalid argument

but if I only use CPU instead of with GPU and CUDA, the program could run. The only problem is TOO SLOW. 

So, if anyone meet the same question and solve it, please tell me the solution. Thank you very much!

**Comment by JackSnowWolf** [comment]
@ralek67 
Here is what I write in my cfg file. I only use one yolo layer since my target is somehow stable. 

[convolutional]
size=1
stride=1
pad=1
filters=18
activation=linear


[yolo]
mask = 0,1,2
anchors = 46, 45, 78, 46, 96, 78
classes=1
num=3
jitter=.05
ignore_thresh = .001
truth_thresh = 0.9
random=0

when I set like this, the program could run rightly. But, when the filters = 24, mask = 0,1,2,3, anchors have eight value. num = 4, the program will raise the error I write before. If without cuda and GPU, the program could run very slowly. So, I guess there might be some default value inside cuda code. Do you have any idea about this? How to revise the code to train normally? 

BTW, I tried to use two [yolo] layers to get a better result. But since my target sizes are similar, the former [yolo] layer and the later [yolo] layer form a kind of competition. Therefore, I want to increase the amount of anchors in one [yolo] layer to avoid this problem. That is my motivation to do this.

PS: I have changed the random scale inside the yolov3.

Hope what I say could help you understand my scenario. If you have any idea about this, please help me! I will appreciate it！

**Comment by ralek67** [comment]
When does the error come? Directly when you try to train?

I've tried on my computer with only one yolo layer:
```
[convolutional]
size=1
stride=1
pad=1
filters=117
activation=linear


[yolo]
mask = 0,1,2,3,4,5,6,7,8
anchors = 10,13,  16,30,  33,23,  30,61,  62,45,  59,119,  116,90,  156,198,  373,326
classes=8
num=9
jitter=.3
ignore_thresh = .5
truth_thresh = 1
random=1
```
And training is working with GPU=1 CUDNN=1

I use CUDA 9.0

**Comment by JackSnowWolf** [comment]
@ralek67 
I used CUDA Version 8.0.61
It is possible that the error comes from the problem of different versions.

The error just came from the begining.

**Comment by ralek67** [comment]
@JackSnowWolf 
I'm not an expert so I can't help you more than that.
What you wrote in your config file is correct so it's strange.

Can't you try to update to CUDA 9.0?

If no you should also try this Darknet fork: https://github.com/AlexeyAB/darknet

**Comment by libzzluo** [comment]
Hi! @ralek67 
It seems that you modify the .cfg file (design a similar new network?) and train the model on your own dataset.
then.. which .weight file do you choose to load? the darknet53 (pretraining on ImageNet)? or just initializing the weight?
